src/cache.rs

```diff
@@ -143,40 +144,65 @@ impl Cache {
     }
 
     /// Search for a page and return the path to it.
-    pub fn find_page(&self, name: &str) -> Option<PathBuf> {
-        // Build page file name
-        let page_filename = format!("{}.md", name);
-
+    pub fn find_pages(&self, name: &str, config: &Config) -> Vec<PathBuf> {
```

Comment:
For performance and readability reasons, I think it would be good if this wouldn't return a `Vec`, but instead two separate values. Additionally, it can be made easier to read by creating a custom struct:

```rust
struct PageLookupResult {
    page_path: PathBuf,
    patch_path: Option<PathBuf>,
}
```

This way, the name of the function can be kept as `find_page`. After all, it finds a page and corresponding patch files, not multiple pages.

```rust
pub fn find_page(&self, name: &str, config: &Config) -> Option<PageLookupResult> {
```

The calling code can still use a `for path in` loop, if you implement a method that returns an iterator:

```rust
impl PageLookupResult {
    fn files(&self) -> impl Iterator<Item = &Path> {
        iter::once(self.page_path.as_path())
            .chain(self.patch_path.as_deref().into_iter())
    }
}
```

See https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6131f92216308d4866fecbe09c8736ef for an interactive version that includes tests.